homeassistant/components/watttime/config_flow.py

```diff
@@ -0,0 +1,237 @@
+"""Config flow for WattTime integration."""
+from __future__ import annotations
+
+import asyncio
+from typing import TYPE_CHECKING, Any
+
+from aiowatttime import Client
+from aiowatttime.errors import (
+    CoordinatesNotFoundError,
+    InvalidCredentialsError,
+    UsernameTakenError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_EMAIL,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+)
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+
+from .const import (
+    AUTH_TYPE_LOGIN,
+    AUTH_TYPE_REGISTER,
+    CONF_BALANCING_AUTHORITY,
+    CONF_BALANCING_AUTHORITY_ABBREV,
+    CONF_BALANCING_AUTHORITY_ID,
+    DOMAIN,
+    LOGGER,
+)
+
+CONF_AUTH_TYPE = "auth_type"
+CONF_ORGANIZATION = "organization"
+
+STEP_LOGIN_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+STEP_REGISTER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(CONF_EMAIL): str,
+        vol.Required(CONF_ORGANIZATION): str,
+    }
+)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_AUTH_TYPE): vol.In([AUTH_TYPE_LOGIN, AUTH_TYPE_REGISTER]),
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WattTime."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        self._client: Client | None = None
+        self._password: str | None = None
+        self._username: str | None = None
+
+    @property
+    def coordinates_schema(self) -> vol.Schema:
+        """Return the coordinates schema."""
+        return vol.Schema(
+            {
+                vol.Required(
+                    CONF_LATITUDE, default=self.hass.config.latitude
```

Comment:
This works, but I also really like what we did for CO2Signal, allow user to pick "Home" as an option, because that's what you want 99% of the time. https://github.com/home-assistant/core/blob/dev/homeassistant/components/co2signal/config_flow.py#L98-L99